### PR TITLE
Bug 1843536 - Fix tab reordering animation issues

### PR DESCRIPTION
**Pull request by Alexandru2909**
Fixed animation issue when switching between two tabs. The problem was caused by the `DragItemContainer` picking up the offset from the last index of the dragged item. Computing the offset based on the index of the item being dragged, instead of the key of the item leads to more consistent animations.

| Before | After |
| --- | --- |
| <video src="https://github.com/mozilla-mobile/firefox-android/assets/35462038/bcd60780-2b8f-458e-aae4-7f3e96893f9a" /> | <video src="https://github.com/mozilla-mobile/firefox-android/assets/35462038/3854fb11-e3cc-4a22-8104-ca18171434ee" /> |

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1843536

### CODE REVIEW

File: fenix/app/src/main/java/org/mozilla/fenix/tabstray/browser/compose/ReorderableGrid.kt [hunk 1]
```diff
@@ -102,11 +102,17 @@ class GridReorderState internal constructor(
...
     private var draggingItemInitialOffset by mutableStateOf(Offset.Zero)
     internal var moved by mutableStateOf(false)
-    internal val draggingItemOffset: Offset
+    private val draggingItemOffset: Offset
         get() = draggingItemLayoutInfo?.let { item ->
             draggingItemInitialOffset + draggingItemCumulatedOffset - item.offset.toOffset()
         } ?: Offset.Zero
 
+    internal fun computeItemOffset(index: Int): Offset {
+        val item = gridState.layoutInfo.visibleItemsInfo.firstOrNull { info -> info.index == index }
```
Comment (t-p-white) [inline fenix/app/src/main/java/org/mozilla/fenix/tabstray/browser/compose/ReorderableGrid.kt:111]: nit: do you think `item` could be more descriptive? e.g. itemAtIndex ?

**Review by t-p-white — APPROVED** [review]
lgtm 🙂

File: fenix/app/src/main/java/org/mozilla/fenix/tabstray/browser/compose/ReorderableList.kt [hunk 2]
```diff
@@ -192,21 +201,23 @@ class ListReorderState internal constructor(
...
  * @param key Key of the item to be displayed.
+ * @param index Position in the list of the item to be displayed.
  * @param content Content of the item to be displayed.
  */
 @ExperimentalFoundationApi
 @Composable
 fun LazyItemScope.DragItemContainer(
     state: ListReorderState,
     key: Any,
+    index: Int,
```
Comment (t-p-white) [inline fenix/app/src/main/java/org/mozilla/fenix/tabstray/browser/compose/ReorderableList.kt:212]: nit: I wonder if index should be called pos or position

**Review by MozillaNoah — APPROVED** [review]
This looks great 😍

**Comment by Alexandru2909** [comment]
Noticed a missing `abs` call when calculating the drag distance in `ReorderableList` during testing. Also noticed `scrollToItem` works better than `scrollBy` when scrolling upwards to the first element in list mode. I wanted to add these minor changes to this PR but since it's already approved I opened another [ticket](https://bugzilla.mozilla.org/show_bug.cgi?id=1852623) in this scope.

**Pull request merged**